Makefile

```diff
@@ -1,17 +1,48 @@
-init: composer-update up
+# Output the help for each task, see https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
+.PHONY: help
+.DEFAULT_GOAL := help
 
-up:
-	docker-compose up -d
-down:
-	docker-compose down
+# Run silent.
+MAKEFLAGS += --silent
 
-composer-update:
-	docker-compose run php composer update
+RUN_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+$(eval $(RUN_ARGS):;@:)
 
-yii3:
-	docker-compose run php ./yii $(filter-out $@, $(MAKECMDGOALS))
+up: ## Up the dev environment.
+	docker compose -f docker-compose.dev.yml up -d
 
-test:
-	docker-compose run php ./vendor/bin/codecept run
-psalm:
-	docker-compose run php ./vendor/bin/psalm
+up-build: ## Up the dev environment rebuilding images.
+	docker compose -f docker-compose.dev.yml up -d --build
+
+down: ## Down the dev environment.
+	docker compose -f docker-compose.dev.yml down
+
+cmd: ## Run a command within the container.
+	docker compose -f docker-compose.dev.yml exec app $(CMD) $(RUN_ARGS)
+
+shell: CMD="/bin/sh" ## Get into container shell.
+shell: cmd
+
+yii: CMD="./yii" ## Execute Yii command.
+yii: cmd
+
+composer: CMD="composer" ## Run Composer.
+composer: cmd
+
+codecept: CMD="./vendor/bin/codecept" ## Run Codeception.
+codecept: cmd
+
+psalm: CMD="./vendor/bin/psalm" ## Run Psalm.
+psalm: cmd
+
+build: ## Build an image.
+	docker build --target prod --pull -t app-api:latest .
```

Comment:
> -t app-api:latest

I'd hide it with `${IMAGE}` and `${IMAGE_TAG}` to change image app-api one time for whole project and build prod containers not only for `latest` tag.